### ISSUE TITLE
ncurses-6.0 fails to build on Ubuntu 16.04

### ISSUE THREAD

**Issue by riccardomurri** [hpcugent/easybuild-easyconfigs#3622]
Trying to build `ncurses-6.0.eb` on a Ubuntu 16.04 VM with the `foss-2016b`
toolchain installed ends with an error::

```
$ eb /opt/easybuild/software/EasyBuild/2.9.0/lib/python2.7/site-packages/easybuild_easyconfigs-2.9.0-py2.7.egg/easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb
== temporary log file in case of crash /tmp/eb-kxD7o6/easybuild-K6tscS.log
== resolving dependencies ...
== processing EasyBuild easyconfig /opt/easybuild/software/EasyBuild/2.9.0/lib/python2.7/site-packages/easybuild_easyconfigs-2.9.0-py2.7.egg/easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb
== building and installing ncurses/6.0...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== FAILED: Installation ended unsuccessfully (build directory: /tmp/ncurses/6.0/dummy-dummy): build failed (first 300 chars): cmd " make -j 1 " exited with exitcode 2 and output:
cd man && make DESTDIR="" RPATH_LIST="/opt/easybuild/software/ncurses/6.0/lib" all
make[1]: Entering directory '/tmp/ncurses/6.0/dummy-dummy/ncurses-6.0/man'
/bin/sh ./MKterminfo.sh ./terminfo.head ./../include/Caps ./terminfo.tail >terminfo.5
mak
== Results of the build can be found in the log file(s) /tmp/eb-kxD7o6/easybuild-ncurses-6.0-20160927.130156.iSdal.log
ERROR: Build of /opt/easybuild/software/EasyBuild/2.9.0/lib/python2.7/site-packages/easybuild_easyconfigs-2.9.0-py2.7.egg/easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb failed (err: 'build failed (first 300 chars): cmd " make -j 1 " exited with exitcode 2 and output:\ncd man && make DESTDIR="" RPATH_LIST="/opt/easybuild/software/ncurses/6.0/lib" all\nmake[1]: Entering directory \'/tmp/ncurses/6.0/dummy-dummy/ncurses-6.0/man\'\n/bin/sh ./MKterminfo.sh ./terminfo.head ./../include/Caps ./terminfo.tail >terminfo.5\nmak')
```

The log file shows this error::

```
gcc -DHAVE_CONFIG_H -I. -I../include  -D_GNU_SOURCE -DNDEBUG -O2 --param max-inline-insns-single=1200 -fPIC -c ../ncurses/lib_gen.c -o ../obj_s/lib_gen.o
In file included from ./curses.priv.h:325:0,
                 from ../ncurses/lib_gen.c:19:
_24717.c:843:15: error: expected ')' before 'int'
../include/curses.h:1631:56: note: in definition of macro 'mouse_trafo'
 #define mouse_trafo(y,x,to_screen) wmouse_trafo(stdscr,y,x,to_screen)
                                                        ^
Makefile:2155: recipe for target '../obj_s/lib_gen.o' failed
make[1]: *** [../obj_s/lib_gen.o] Error 1
make[1]: Leaving directory '/tmp/ncurses/6.0/dummy-dummy/ncurses-6.0/ncurses'
Makefile:113: recipe for target 'all' failed
make: *** [all] Error 2
 (at easybuild/software/EasyBuild/2.9.0/lib/python2.7/site-packages/easybuild_framework-2.9.0-py2.7.egg/easybuild/tools/run.py:438 in parse_cmd_output)
```

This is a kind of showstopper since `ncurses` is a dependency of many software packages ...

**Comment by riccardomurri** [comment]
Looks like some more patching/workaround is needed: http://stackoverflow.com/questions/37475222/ncurses-6-0-compilation-error-error-expected-before-int

**Comment by boegel** [comment]
@riccardomurri This is probably caused by having GCC 5.x as a system compiler; can you try adding this to the `ncurses-6.0.eb` easyconfig (the patch is already included in EasyBuild v2.9.0, it's used in other `ncurses` easyconfigs where a GCC 5.x-based toolchain is used):

```python
patches = ['ncurses-6.0_gcc-5.patch']
```

cfr. https://github.com/hpcugent/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/n/ncurses/ncurses-6.0_gcc-5.patch

**Comment by riccardomurri** [comment]
It's a `mawk` vs `gawk` problem; fix in PR #3624

**Comment by boegel** [comment]
follow-up in #3624
